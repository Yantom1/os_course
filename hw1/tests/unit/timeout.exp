smash> smash error: timeout: invalid arguments
smash> smash error: timeout: invalid arguments
smash> smash error: timeout: invalid arguments
smash> my_sleep
random1.txt
random2.txt
sub1
sub2
sub3
smash> before
smash> smash: got an alarm
after
smash> 0
Done: 2
smash> smash: got an alarm
after
smash> 0
2
smash: got an alarm
smash: timeout 3 ./my_sleep 4 timed out!
smash> smash: got ctrl-C
smash> smash> 0
\[1\] timeout 3 ./my_sleep 4 & : \d+ \d+ secs
smash> 2
smash: got an alarm
smash: timeout 3 ./my_sleep 4 & timed out!
smash: got ctrl-C
smash> 0
2
smash: got ctrl-C
smash: process \d+ was killed
smash> before
smash> smash: got an alarm
after
smash> smash> smash> 0
0
2
2
4
smash: got an alarm
smash: timeout 5 ./my_sleep 10 & timed out!
4
6
8
smash: got an alarm
smash: timeout 9 ./my_sleep 20 timed out!
smash> smash> 0
smash> 0
2
2
4
smash: got an alarm
smash: timeout 3 ./my_sleep 20& timed out!
6
smash: got an alarm
smash: timeout 7 ./my_sleep 10& timed out!
smash: got ctrl-C
smash> 0
smash error: bg: there is no stopped jobs to resume
smash> smash: got ctrl-Z
smash error: bg: there is no stopped jobs to resume
smash> 2
smash: got an alarm
smash: timeout 3 ./my_sleep 20& timed out!
smash> smash> 0
timeout 3 ./my_sleep 20& : \d+
smash: got ctrl-Z
smash: process \d+ was stopped
smash> timeout 3 ./my_sleep 20& : \d+
smash> 2
smash: got an alarm
smash: timeout 3 ./my_sleep 20& timed out!
smash> smash> 0
timeout 3 ./my_sleep 20& : \d+
smash: got ctrl-C
smash: process \d+ was killed
smash> smash> smash: got an alarm
smash> smash> smash> smash> \[1\] timeout 5 sleep 100& : \d+ \d+ secs
\[2\] timeout 10 sleep 200& : \d+ \d+ secs
\[3\] timeout 2 sleep 300& : \d+ \d+ secs
\[4\] timeout 7 sleep 400& : \d+ \d+ secs
smash> smash: got an alarm
smash: timeout 2 sleep 300& timed out!
smash: got an alarm
smash: timeout 5 sleep 100& timed out!
smash: got an alarm
smash: timeout 7 sleep 400& timed out!
smash: got an alarm
smash: timeout 10 sleep 200& timed out!
smash> smash> smash: got an alarm
smash: timeout 2 sleep 4& timed out!
smash: got ctrl-C
smash: process \d+ was killed
smash> smash> smash: got an alarm
smash: got ctrl-C
smash: process \d+ was killed
smash> smash: sending SIGKILL signal to 0 jobs:
