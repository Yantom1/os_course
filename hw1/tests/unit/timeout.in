# First invalid args
timeout
timeout 4
timeout -4 showpid
#Check command finished before alarm
timeout 3 ls
!time.sleep(2)
echo before
!time.sleep(2)
# Got alarm should be print now, after 2 jobs
echo after
# Now check with &
#Check command finished before alarm
timeout 3 ./my_sleep 2
!time.sleep(4)
# Got alarm should be print now, after done
echo after
# Now we let the alarm to kill
timeout 3 ./my_sleep 4
!time.sleep(4)
# Make sure we didnt kill anyone
CtrlC
jobs
# Now with &
timeout 3 ./my_sleep 4 &
jobs
!time.sleep(4)
# Should be empty
# Make sure we didnt kill anyone
CtrlC
jobs
# Now kill in middle of action
timeout 6 ./my_sleep 4
!time.sleep(3)
CtrlC
echo before
!time.sleep(4)
# Now should print got alarm
echo after
jobs
timeout 5 ./my_sleep 10 &
!time.sleep(1)
timeout 9 ./my_sleep 20
!time.sleep(10)
timeout 7 ./my_sleep 10&
!time.sleep(1)
timeout 3 ./my_sleep 20&
!time.sleep(7)
CtrlC
timeout 3 ./my_sleep 20&
bg
CtrlZ
bg
!time.sleep(4)
jobs
timeout 3 ./my_sleep 20&
fg
CtrlZ
bg
!time.sleep(4)
jobs
timeout 3 ./my_sleep 20&
fg
CtrlC
jobs
!time.sleep(4)
timeout 5 sleep 100&
timeout 10 sleep 200&
timeout 2 sleep 300&
timeout 7 sleep 400&
jobs
!time.sleep(11)
jobs
#
# Get alarm during other command
#
timeout 2 sleep 4&
sleep 10
!time.sleep(3)
# The alarm should kill the sleep4& 
# Kill sleep 10
CtrlC
# Now we check in case the command died before alarm and during other command (Checking that you clean jobs list in alarm handler as well)
timeout 3 sleep 1&
sleep 10
!time.sleep(5)
# Should kill the sleep4& 
CtrlC
quit kill
