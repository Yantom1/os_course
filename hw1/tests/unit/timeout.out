smash> smash error: timeout: invalid arguments
smash> smash error: timeout: invalid arguments
smash> smash error: timeout: invalid arguments
smash> my_sleep
random1.txt
random2.txt
sub1
sub2
sub3
smash> before
smash> smash: got an alarm
after
smash> 0
Done: 2
smash> smash: got an alarm
after
smash> 0
2
smash: got an alarm
smash: timeout 3 ./my_sleep 4 timed out!
smash> smash: got ctrl-C
smash> smash> 0
[1] timeout 3 ./my_sleep 4 & : 45366 0 secs
smash> 2
smash: got an alarm
smash: timeout 3 ./my_sleep 4 & timed out!
smash: got ctrl-C
smash> 0
2
smash: got ctrl-C
smash: process 45373 was killed
smash> before
smash> smash: got an alarm
after
smash> smash> smash> 0
0
2
2
4
smash: got an alarm
smash: timeout 5 ./my_sleep 10 & timed out!
4
6
8
smash: got an alarm
smash: timeout 9 ./my_sleep 20 timed out!
smash> smash> 0
smash> 0
2
2
4
smash: got an alarm
smash: timeout 3 ./my_sleep 20& timed out!
6
smash: got an alarm
smash: timeout 7 ./my_sleep 10& timed out!
smash: got ctrl-C
smash> 0
smash error: bg: there is no stopped jobs to resume
smash> smash: got ctrl-Z
smash error: bg: there is no stopped jobs to resume
smash> 2
smash: got an alarm
smash: timeout 3 ./my_sleep 20& timed out!
smash> smash> 0
timeout 3 ./my_sleep 20& : 45413
smash: got ctrl-Z
smash: process 45413 was stopped
smash> timeout 3 ./my_sleep 20& : 45413
smash> 2
smash: got an alarm
smash: timeout 3 ./my_sleep 20& timed out!
smash> smash> 0
timeout 3 ./my_sleep 20& : 45421
smash: got ctrl-C
smash: process 45421 was killed
smash> smash> smash: got an alarm
smash> smash> smash> smash> [1] timeout 5 sleep 100& : 45427 0 secs
[2] timeout 10 sleep 200& : 45429 0 secs
[3] timeout 2 sleep 300& : 45431 0 secs
[4] timeout 7 sleep 400& : 45433 0 secs
smash> smash: got an alarm
smash: timeout 2 sleep 300& timed out!
smash: got an alarm
smash: timeout 5 sleep 100& timed out!
smash: got an alarm
smash: timeout 7 sleep 400& timed out!
smash: got an alarm
smash: timeout 10 sleep 200& timed out!
smash> smash> smash: got an alarm
smash: timeout 2 sleep 4& timed out!
smash: got ctrl-C
smash: process 45443 was killed
smash> smash> smash: got an alarm
smash: got ctrl-C
smash: process 45448 was killed
smash> smash: sending SIGKILL signal to 0 jobs:
